Src/Support/Google.Apis.Auth.Tests/OAuth2/ImpersonatedCredentialTests.cs

```diff
@@ -0,0 +1,230 @@
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        private ImpersonatedCredential CreateImpersonatedCredentialWithSignBlobResponse()
+        {
+            var sourceCredential = CreateSourceCredential();
+            // Use signedBlob = base64("foo") = "Zm9v"
+            var body = new { keyId = "1", signedBlob = "Zm9v" };
+            var content = NewtonsoftJsonSerializer.Instance.Serialize(body);
+            var messageHandler = new MockHttpMessageHandler(HttpStatusCode.OK, content);
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        private ImpersonatedCredential CreateImpersonatedCredentialWithErrorResponse()
+        {
+            var sourceCredential = CreateSourceCredential();
+            var content = errorResponseContent;
+            var messageHandler = new MockHttpMessageHandler(HttpStatusCode.NotFound, content);
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo", new string[] {"scope"})
+            {
+                Clock = clock,
+                HttpClientFactory = new MockHttpClientFactory(messageHandler)
+            };
+            return new ImpersonatedCredential(initializer);
+        }
+
+        [Fact]
+        public void TestConstructorInvalidSourceCredential()
+        {
+            var sourceCredential = GoogleCredential.FromComputeCredential();
+            var initializer = new ImpersonatedCredential.Initializer(sourceCredential, "foo");
+            var ex = Assert.Throws<InvalidOperationException>(() => new ImpersonatedCredential(initializer));
+            Assert.Equal("The underlying credential of source credential must be UserCredential or ServiceAccountCredential.", ex.Message);
+        }
+
+        [Fact]
+        public void TestConstructorMissingTargetPrincipal()
+        {
+            var initializer = new ImpersonatedCredential.Initializer(CreateSourceCredential(), "");
+            Assert.Throws<ArgumentException>(() => new ImpersonatedCredential(initializer));
+        }
+
+        [Fact]
+        public void TestConstructorInvalidLifetime()
+        {
+            var initializer = new ImpersonatedCredential.Initializer(CreateSourceCredential(), "foo", null, null, 50000);
+            var ex = Assert.Throws<InvalidOperationException>(() => new ImpersonatedCredential(initializer));
+            Assert.Equal("Lifetime must be less than or equal to 43200.", ex.Message);
```

Comment:
I'd suggest just asserting that the message *starts* with "Lifetime must be less than or equal to". We don't want to have to change the test if the limit changes.

Reply:
done